packages/dashboard-message-bus-common/lib/messages.ts

```diff
@@ -25,9 +27,18 @@ export interface DashboardProviderMessage extends Message {
   };
 }
 
+export type WorkflowCompileResultMessageType = "workflow-compile-result";
+export const workflowCompileResultMessageType = "workflow-compile-result";
+/**
+ * Message to inform subscribers of new WorkflowCompileResult
+ */
+export interface WorkflowCompileResultMessage extends Message {
+  type: WorkflowCompileResultMessageType;
+  payload: WorkflowCompileResult;
+}
```

Comment:
I think it's incohesive to define such a specific message type here.

The message bus common package is meant to house logic and types that are common between the message bus service and the message bus client. In general I think it makes sense to define a generic message envelope in here, but I think defining the structure of messages that are specific to some other package (compile-common in this case) is probably a code smell.

The easy test for why it's problematic is that you need to take a dependency on `compile-common` here. Perhaps this could be avoided by defining a more generic and extensible message envelope type here that you could use to embed this same information? For example, for messages to do with RPC calls we just have the `DashboardProviderMessage` type. We don't have one message type per type of RPC call, however.

Perhaps there's a more generic `CLIEventMessageType` structure that would work better here?

If you aren't familiar with how other message busses work, it might be worth having a look at the message envelope types used by MQTT and AMQP.

Reply:
> In general I think it makes sense to define a generic message envelope in here, but I think defining the structure of messages that are specific to some other package (compile-common in this case) is probably a code smell.

Put differently - there's no reason why any of the core message bus service logic _must_ know anything at all about `WorkflowCompileResult` in order do its job effectively. It should be able to treat that as opaque data that it's just shuffling around.

Reply:
Thanks Ben. Initially I was uncertain about having an envelope message type that represents any CLI event data, mostly because the payload must necessarily be typed `any`. But I figured that I can just type it `WorkflowCompileResult`  in the frontend, which admittedly isn't as nice as the given type guarantees ensured by `isWorkflowCompileResultMessage`, but this is a worthwhile tradeoff compared to taking `compile-common` as a dependency and making more specific message types in the future as dashboard grows, which does smell.

I've made [updates](https://github.com/trufflesuite/truffle/commit/35a9980fa901c2d1add466f0927541ae44d8abed) to use `CliEventMessage` and got rid of `WorkflowCompileResultMessage`.